### PR TITLE
New Cliff Lorimer quantification

### PR DESCRIPTION
**Pull request by pburdet**
Almost ready to merge. For this function to be used properly, It needs to a remove_background specific to EDS_TEM. Then I will write a proper documentation.

### CODE REVIEW

**Comment by to266** [comment]
======================================================================
ERROR: hyperspy.tests.utils.test_material.TestWeightToFromAtomic.test_multi_dim
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/to266/anaconda/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/to266/hyperspy/hyperspy/tests/utils/test_material.py", line 36, in test_multi_dim
    at = hs.utils.material.weight_to_atomic(wt, elements)
  File "/home/to266/hyperspy/hyperspy/misc/material.py", line 74, in weight_to_atomic
    atomic_percent = stack(weight_percent)
  File "/home/to266/hyperspy/hyperspy/misc/utils.py", line 723, in stack
    original_shape = obj.data.shape
AttributeError: 'buffer' object has no attribute 'shape'

======================================================================
ERROR: hyperspy.tests.utils.test_material.test_density_of_mixture
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/to266/anaconda/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/to266/hyperspy/hyperspy/tests/utils/test_material.py", line 60, in test_density_of_mixture
    wt, elements)[0, 0])
  File "/home/to266/hyperspy/hyperspy/misc/material.py", line 235, in density_of_mixture_of_pure_elements
    density = weight_percent[0].deepcopy()
AttributeError: 'numpy.ndarray' object has no attribute 'deepcopy'

----------------------------------------------------------------------
Ran 618 tests in 15.575s

FAILED (errors=2)

File: hyperspy/_signals/eds_tem.py [hunk 1]
```diff
@@ -242,3 +245,171 @@ def get_calibration_from(self, ref, nb_pix=1):
...
+            The mask of the region
+        """
+        from scipy.ndimage.morphology import binary_dilation, binary_erosion
+        mask = (self.max(-1) <= threshold)
+        if closing:
+            mask.data = binary_dilation(mask.data, border_value=0)
+            mask.data = binary_erosion(mask.data, border_value=1)
+        return mask
+
+    def decomposition(self,
```
Comment (to266) [inline hyperspy/_signals/eds_tem.py]: Shouldn't it be moved to a separate PR? Otherwise I think it's fine

File: hyperspy/_signals/eds_tem.py [hunk 2]
```diff
@@ -242,3 +245,171 @@ def get_calibration_from(self, ref, nb_pix=1):
...
+        >>> res = s.quantification_cliff_lorimer(intensities,kfactors)
+
+        See also
+        --------
+        vacuum_mask
+        """
+        if isinstance(navigation_mask, float):
+            navigation_mask = self.vacuum_mask(navigation_mask, closing).data
+        elif navigation_mask is not None:
+            navigation_mask = navigation_mask.data
```
Comment (to266) [inline hyperspy/_signals/eds_tem.py]: That means the `navigation_mask` is a signal? In the docs it says it's a np.array....

File: doc/user_guide/eds.rst [hunk 3]
```diff
@@ -348,3 +348,31 @@ an interactive way to generate intensity map.
...
+For the moment, only one TEM quantification method (Cliff-Lorimer) is implemented.
+
+Quantification can be applied from the intensities (background subtracted) with the :py:meth:`~._signals.eds_tem.EDSTEMSpectrum.quantification_cliff_lorimer` method. The required kfactors can be usually found in the EDS manufacturer software.
+
+.. code-block:: python
+
+    >>> spec.set_elements(["Al", "Cr", "Ni"])
+    >>> spec.add_lines()
+    >>> kfactors = [0.982, 1.32, 1.60]
+    >>> weight_percent = spec.quantification_cliff_lorimer(intensities, kfactors)
```
Comment (to266) [inline doc/user_guide/eds.rst]: Maybe add a line in the example how to get the `intensities`?

File: hyperspy/_signals/eds_tem.py [hunk 4]
```diff
@@ -242,3 +245,171 @@ def get_calibration_from(self, ref, nb_pix=1):
...
+        Return
+        ------
+        mask: signal
+            The mask of the region
+        """
+        from scipy.ndimage.morphology import binary_dilation, binary_erosion
+        mask = (self.max(-1) <= threshold)
+        if closing:
+            mask.data = binary_dilation(mask.data, border_value=0)
+            mask.data = binary_erosion(mask.data, border_value=1)
```
Comment (to266) [inline hyperspy/_signals/eds_tem.py]: This closing only gets rid of `False` holes in the mask (i.e. a single `False` pixel, surrounded by `True` on most sides). Is it intended to leave the `True` holes? Two more reversed identical operations would get rid of them (i.e. first dilation, then erosion, so that the whole treatment would look like dilation -> erosion -> erosion -> dilation)

**Comment by to266** [comment]
Also, I am not sure which PR it is related to (apologies if not this one / not yours), but in `get_lines_intensity` (of `EDSTEMSpectrum`) documentation, energy resolution refers to SEM twice (one, I guess, should be changed to TEM)

File: doc/user_guide/eds.rst (hunk 3 above)
Comment (pburdet) [inline doc/user_guide/eds.rst]: I'm waiting PR #470 to get the complete example...

File: hyperspy/_signals/eds_tem.py (hunk 1 above)
Comment (pburdet) [inline hyperspy/_signals/eds_tem.py]: It is PR #407. Sorry for the confusion

File: hyperspy/_signals/eds_tem.py (hunk 2 above)
Comment (pburdet) [inline hyperspy/_signals/eds_tem.py]: Again wrong PR (see #407). True I will udpate the doc

File: hyperspy/_signals/eds_tem.py (hunk 4 above)
Comment (pburdet) [inline hyperspy/_signals/eds_tem.py]: Done

**Comment by pburdet** [comment]
tests are working and last comment corrected

**Comment by pburdet** [comment]
- Fast_forwarded
- Rename the main function, more general. In the future, we may have more than one quantification method
- Cleaning of the code (pep8)
